addons/website_sale/static/tests/tours/website_sale_category_page_and_products_snippet.js

```diff
@@ -0,0 +1,59 @@
+/** @odoo-modules */
+
+import tour from 'web_tour.tour';
+import wTourUtils from 'website.tour_utils';
+
+const PRODUCT_CATEGORY_ID = 2;
+
+wTourUtils.registerWebsitePreviewTour('category_page_and_products_snippet_edition', {
+    test: true,
+    url: `/shop/category/${PRODUCT_CATEGORY_ID}`,
+    edition: true,
+}, [
+    Object.assign(wTourUtils.dragNDrop({id: 's_dynamic_snippet_products', name: 'Products'}), {
+        content: "Drag and drop the product snippet inside the category area",
+        run: 'drag_and_drop iframe #category_header',
+    }),
+    {
+        content: "Click on the product snippet to show its options",
+        trigger: 'iframe #category_header .s_dynamic_snippet_products',
+    },
+    {
+        content: "Open category option dropdown",
+        trigger: 'we-select[data-attribute-name="productCategoryId"] we-toggler',
+    },
+    {
+        content: "Choose the option to use the current page's category",
+        trigger: 'we-button[data-select-data-attribute="current"]',
```

Comment:
Shouldn't there be a check step after this and before click on save to prevent race condition and wait for the option click action?

Reply:
I don't think so, it would be counterproductive IMO: more code, less efficient and testing less stuff:
- "prevent race condition" -> well if there is one, it should be fixed code-wise :fearful: Not hidden by the test. But there should not be any anyway, clicking on a snippet option then immediately on save is normally handled correctly. The only problematic case (probably outside some text-editor-related stuff) is drag and dropping a snippet then immediately saving (which does not scare me too much right now as really not a normal user use case). Using snippet options is normally robust.
- Checking if the option really did have an effect is handled by the next tour in that test :)